api/proto/teleport/legacy/client/proto/authservice.proto

```diff
@@ -1153,6 +1153,29 @@ message ListRolesResponse {
   string NextKey = 2;
 }
 
+// ListRequestableRolesRequest is a request to list roles that the user can request.
+message ListRequestableRolesRequest {
+  // Limit is the maximum amount of roles per page.
+  int32 Limit = 1;
+
+  // StartKey is used to resume a query in order to enable pagination.
+  // If the previous response had NextKey set then this should be
+  // set to its value. Otherwise leave empty.
+  string StartKey = 2;
+
+  // Filter matches roles.
+  types.RoleFilter Filter = 3;
+}
+
+// ListRequestableRolesResponse is a page of requestable roles.
+message ListRequestableRolesResponse {
+  // Roles is a page of roles that the user can request.
+  repeated types.RoleV6 Roles = 1;
```

Comment:
Curious why and where do we need the whole role type? Web UI only needs requestable role names?

Reply:
Good catch, there's no particular reason, it was just readily available and allowed us to reuse the WebUI types for listing roles instead of creating new ones, it also allowed us to add a `Description` column in the table. However now that you mention it I realize that sending over all the details of every requestable role to the client might not be a good approach security-wise. I'll trim this to send only the role name and description.